main.rb

```diff
@@ -29,17 +30,61 @@ def choose_number(choice, app)
 end
 
 def main
-  app = App.new
+  # load the data and then parse it to the app
 
+  books = load_book_data
+  persons = load_persons_data
+  rentals = load_rental_data(persons, books)
+  app = App.new(books, persons, rentals)
   loop do
     render_choices
     choice = gets.chomp.to_i
     if choice >= 7
+      # we will save the data
+      app.store_data
       puts 'Thank you for using this app'
       break
     end
     choose_number(choice, app)
   end
 end
 
+def load_book_data
+  all_book = []
+  if File.exist?('data/books.json') && !File.empty?('data/books.json')
+    book_data = JSON.parset(File.read('data/books.json'))
```

Comment:
- [ ] There is an error here and your app cant start with it, it should be `JSON.parse` you need to remove the t in front of it.